app/src/main/java/com/example/kotlincalculator/CalculatorActivity.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.kotlincalculator
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import com.example.kotlincalculator.mvp.model.CalculatorModel
+import com.example.kotlincalculator.mvp.presenter.CalculatorPresenter
+import com.example.kotlincalculator.mvp.view.CalculatorView
+import kotlinx.android.synthetic.main.activity_main.*
+
+class CalculatorActivity : AppCompatActivity(), View.OnClickListener {
+
+    var presenter = CalculatorPresenter(CalculatorView(this), CalculatorModel())
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        this.setListeners()
+    }
+
+    private fun setListeners(){
+        button_zero.setOnClickListener(this)
+        button_one.setOnClickListener(this)
+        button_two.setOnClickListener(this)
+        button_three.setOnClickListener(this)
+        button_four.setOnClickListener(this)
+        button_five.setOnClickListener(this)
+        button_six.setOnClickListener(this)
+        button_seven.setOnClickListener(this)
+        button_eight.setOnClickListener(this)
+        button_nine.setOnClickListener(this)
+        button_sum.setOnClickListener(this)
+        button_subtract.setOnClickListener(this)
+        button_multiply.setOnClickListener(this)
+        button_share.setOnClickListener(this)
+    }
+
+    override fun onClick(view: View?) {
+        val textButton : String = (view as Button).text.toString()
```

Comment:
maybe you can use a more descriptive name for variables

Reply:
done